realm/realm-library/src/main/cpp/io_realm_mongodb_App.cpp

```diff
@@ -241,3 +242,60 @@ JNIEXPORT void JNICALL Java_io_realm_mongodb_App_nativeSwitchUser(JNIEnv* env,
     CATCH_STD()
 }
 
+JNIEXPORT jobject JNICALL Java_io_realm_mongodb_App_nativeMakeStreamingRequest(JNIEnv* env,
+                                                                  jclass,
+                                                                  jlong j_app_ptr,
+                                                                  jlong j_user_ptr,
+                                                                  jstring j_function_name,
+                                                                  jstring j_bson_args,
+                                                                  jstring j_service_name)
+{
+    try {
+        auto app = *reinterpret_cast<std::shared_ptr<App>*>(j_app_ptr);
+        auto user = *reinterpret_cast<std::shared_ptr<SyncUser>*>(j_user_ptr);
+
+        JStringAccessor function_name(env, j_function_name);
+        JStringAccessor service_name(env, j_service_name);
+
+        bson::BsonArray filter(JniBsonProtocol::parse_checked(env, j_bson_args, Bson::Type::Array, "BSON filter must be an Array"));
+
+        const Request &request = app->make_streaming_request(user, function_name, filter,
+                                                             std::string(service_name));
+
+        jstring j_method;
+
+        switch (request.method){
+            case HttpMethod::get:
+                j_method = env->NewStringUTF("get");
+                break;
+            case HttpMethod::post:
+                j_method = env->NewStringUTF("post");
+                break;
+            case HttpMethod::patch:
+                j_method = env->NewStringUTF("patch");
+                break;
+            case HttpMethod::put:
+                j_method = env->NewStringUTF("put");
+                break;
+            case HttpMethod::del:
+                j_method = env->NewStringUTF("del");
+                break;
+        }
+
+        jstring j_url = env->NewStringUTF(request.url.c_str());
+        jobject j_headers = JniUtils::to_hashmap(env, request.headers);
+        jstring j_body = env->NewStringUTF(request.body.c_str());
+
+        jclass request_class = env->FindClass("io/realm/internal/objectstore/OsJavaNetworkTransport$Request");
```

Comment:
We shouldn't look this up on every request. If you make it static it only happens once